### PR TITLE
OTWO-5086 Use fisbot api for code_locations and repositories

### PR DESCRIPTION
**Pull request by notalex**

### CODE REVIEW

**Comment by karthikeyanBD** [comment]
retest this please

**Comment by notalex** [comment]
Build is failing due to vulnerabilities reported by bundle-audit. One of the upgrades is rubocop, which requires a ton of new syntax fixes. I propose that we fix bundle-audit warnings in another ticket.

File: app/controllers/concerns/project_filters.rb [hunk 1]
```diff
@@ -44,8 +44,9 @@ def find_projects_by_params
   def find_forge_matches
     @match = Forge::Match.first(params[:codelocation])
     return unless @match
-    @projects = Project.where(id: Repository.matching(@match).joins(code_locations: :projects)
-                                    .select('projects.id')).not_deleted
+    matching_code_location_ids = CodeLocationSubscription.code_location_ids_matching_forge(@match)
+    matching_enlistments = Enlistment.where(code_location_id: matching_code_location_ids)
+    @projects = Project.where(id: matching_enlistments.pluck(:project_id)).not_deleted
```
Comment (karthikeyanBD) [inline app/controllers/concerns/project_filters.rb]: We can simplify this to single query
`Project.joins(:enlistments).where(enlistments: { code_location_id: matching_code_location_ids}).not_deleted`

File: app/lib/fisbot/code_location_subscription.rb [hunk 2]
```diff
@@ -1,6 +1,34 @@
 class CodeLocationSubscription < FisbotApi
-  def initialize(code_location_id)
-    @endpoint = 'subscriptions'
-    @data = { code_location_id: code_location_id }
+  class << self
+    def endpoint
+      :subscriptions
```
Comment (karthikeyanBD) [inline app/lib/fisbot/code_location_subscription.rb]: :+1:

File: app/lib/fisbot/code_location_subscription.rb [hunk 3]
```diff
@@ -1,6 +1,34 @@
...
+    end
+
+    def code_locations_for_project(project_id)
+      uri = URI("#{api_uri}/api/v1/subscriptions/code_locations/#{project_id}.json?api_key=#{api_key}")
```
Comment (karthikeyanBD) [inline app/lib/fisbot/code_location_subscription.rb]: We could move all these API related routes to some constant file and use some wrapper to just grep and replace, key with a value.

File: app/admin/slave_logs_admin.rb [hunk 4]
```diff
@@ -19,8 +19,7 @@
   controller do
     def scoped_collection
       if params[:code_location_id]
-        code_location = CodeLocation.find(params[:code_location_id])
-        code_location.slave_logs
+        Job.where(code_location_id: :code_location_id).slave_logs
```
[the remark this answers is not in the document]
Comment (notalex) [reply app/admin/slave_logs_admin.rb]: Yes, I did.

File: app/lib/fisbot/code_location_subscription.rb (hunk 3 above)
Comment (notalex) [reply app/lib/fisbot/code_location_subscription.rb]: We moved the routes to another class, greatly simplifying the url generation.

File: app/controllers/concerns/project_filters.rb (hunk 1 above)
Comment (notalex) [reply app/controllers/concerns/project_filters.rb]: Good point

**Comment by karthikeyanBD** [comment]
LGTM

**Pull request merged**